### PR TITLE
fix: adaptive logo shows on more search results

### PR DESCRIPTION
**Pull request by SHolleworth**
4875:

Screenshot 2023-04-14 at 11 51 07

Screenshot 2023-04-14 at 11 51 41

Screenshot 2023-04-14 at 11 52 15

Screenshot 2023-04-14 at 11 52 46

### CODE REVIEW

File: src/workspace/src/home/index.ts [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
+
+export async function registerHome(): Promise<HomeRegistration> {
   const queryMinLength = 3
   const loadingResult = {
-    key: 'loading',
-    title: 'Searching...',
+    key: "loading",
+    title: "Searching...",
     data: {},
+    icon: ADAPTIVE_LOGO,
```
Comment (algreasley) [inline src/workspace/src/home/index.ts:30]: not sure what this fixes .. ?  (we also have icon on cliProvider conf below ..)

File: src/workspace/src/home/index.ts [hunk 2]
```diff
@@ -132,16 +140,16 @@ export async function registerHome(): Promise<void> {
...
+      console.warn("Unable to execute result without data being passed")
     }
   }
 
   const cliProvider: CLIProvider = {
-    title: 'Adaptive Workspace',
+    title: "Adaptive Workspace",
     id: PROVIDER_ID,
-    icon: `${BASE_URL}/images/icons/adaptive.png`,
+    icon: ADAPTIVE_LOGO,
```
Comment (algreasley) [inline src/workspace/src/home/index.ts:150]: no effective change here, ofc, but do you know which part of the WS UI this affects?

File: src/workspace/src/home/nlpProvider.ts [hunk 3]
```diff
@@ -168,30 +194,31 @@ const constructSpotResult = (price: Price, currencyPair: CurrencyPair) => {
   return {
     key: `spot-${symbol}`,
     title: symbol,
-    label: 'Currency Pair',
+    label: "Currency Pair",
+    icon: ADAPTIVE_LOGO,
```
Comment (algreasley) [inline src/workspace/src/home/nlpProvider.ts:198]: found it ! 😄

File: src/workspace/src/home/utils.ts [hunk 4]
```diff
@@ -1,40 +1,47 @@
...
+import { getApps } from "../apps"
+import { getPages } from "../browser"
+import { BASE_URL } from "../consts"
 
-export const HOME_ACTION_DELETE_PAGE = 'Delete Page'
-export const HOME_ACTION_LAUNCH_PAGE = 'Launch Page'
+export const HOME_ACTION_DELETE_PAGE = "Delete Page"
+export const HOME_ACTION_LAUNCH_PAGE = "Launch Page"
+
+export const ADAPTIVE_LOGO = `${BASE_URL}/images/icons/adaptive.png`
```
Comment (SHolleworth) [inline src/workspace/src/home/utils.ts:14]: Variable for the logo url

File: src/workspace/src/home/utils.ts [hunk 5]
```diff
@@ -52,17 +59,18 @@ const mapAppEntriesToSearchEntries = (apps: App[]): CLISearchResult<any>[] =>
...
       actions: [
-        { name: HOME_ACTION_DELETE_PAGE, hotkey: 'CmdOrCtrl+D' },
-        { name: HOME_ACTION_LAUNCH_PAGE, hotkey: 'Enter' }
+        { name: HOME_ACTION_DELETE_PAGE, hotkey: "CmdOrCtrl+D" },
+        { name: HOME_ACTION_LAUNCH_PAGE, hotkey: "Enter" },
       ],
-      data: { tags: ['page'], pageId: page.pageId },
-      template: CLITemplate.Plain
+      data: { tags: ["page"], pageId: page.pageId },
+      icon: ADAPTIVE_LOGO,
```
Comment (SHolleworth) [inline src/workspace/src/home/utils.ts:72]: applied logo to all the search result templates as such

File: src/workspace/src/home/nlpProvider.ts [hunk 6]
```diff
@@ -168,30 +194,31 @@ const constructSpotResult = (price: Price, currencyPair: CurrencyPair) => {
...
     data: {
       symbol,
-      manifestType: 'url',
-      manifest: `${VITE_RT_URL}/spot/${symbol}`
+      manifestType: "url",
+      manifest: `${VITE_RT_URL}/spot/${symbol}`,
```
Comment (algreasley) [inline src/workspace/src/home/nlpProvider.ts:202]: a-ha - didn't _spot_ this before .. should have changed along with `utils.ts`

File: src/workspace/src/home/nlpProvider.ts [hunk 7]
```diff
@@ -429,23 +474,24 @@ export const getNlpResults = async (
...
-        title: 'Market',
-        label: 'Live Rates',
+        title: "Market",
+        label: "Live Rates",
+        icon: ADAPTIVE_LOGO,
         data: {
-          manifestType: 'url',
-          manifest: `${VITE_RT_URL}/tiles`
+          manifestType: "url",
+          manifest: `${VITE_RT_URL}/tiles`,
```
Comment (algreasley) [inline src/workspace/src/home/nlpProvider.ts:482]: damn, another dodgy url ..

File: src/workspace/src/home/nlpProvider.ts [hunk 8]
```diff
@@ -462,32 +508,35 @@ export const getNlpResults = async (
...
           key: `trade-${trade.tradeId}`,
           title: `${trade.tradeId}`,
-          label: 'Trade',
+          label: "Trade",
+          icon: ADAPTIVE_LOGO,
           data: {
-            manifestType: 'url',
-            manifest: `${VITE_RT_URL}/blotter`
+            manifestType: "url",
+            manifest: `${VITE_RT_URL}/blotter`,
```
Comment (algreasley) [inline src/workspace/src/home/nlpProvider.ts:524]: and again

File: src/workspace/src/home/nlpProvider.ts [hunk 9]
```diff
@@ -594,30 +647,31 @@ export const getNlpResults = async (
...
               key,
               title: `Trade ${trade.status}`,
-              label: 'Trade Execution',
+              label: "Trade Execution",
+              icon: ADAPTIVE_LOGO,
               data: {
-                manifestType: 'url',
-                manifest: `${VITE_RT_URL}/blotter`
+                manifestType: "url",
+                manifest: `${VITE_RT_URL}/blotter`,
```
Comment (algreasley) [inline src/workspace/src/home/nlpProvider.ts:654]: again

File: src/workspace/src/utils.ts [hunk 10]
```diff
@@ -11,7 +12,7 @@ export const getSpotTileApps = async (): Promise<App[]> => {
     manifestType: "url",
     manifest: `${VITE_RT_URL}/fx-spot/${symbol}`,
```
Comment (algreasley) [inline src/workspace/src/utils.ts:13]: this is the correct format for RT FX section URLs - sorry, this is my fault for not doing a global replace earlier 😞

File: src/workspace/src/utils.ts [hunk 11]
```diff
@@ -11,7 +12,7 @@ export const getSpotTileApps = async (): Promise<App[]> => {
...
     title: `${symbol} Spot Tile`,
-    icons: [],
+    icons: [{ src: ADAPTIVE_LOGO }],
```
Comment (algreasley) [inline src/workspace/src/utils.ts:15]: nice, I was hoping that it was as simple as this, once we found the icon in the types

File: src/workspace/src/home/index.ts (hunk 1 above)
Comment (SHolleworth) [reply src/workspace/src/home/index.ts:30]: There's result specifically for the loading state, by default it showed an S like markets showed an M

Comment (algreasley) [reply src/workspace/src/home/index.ts:30]: ah, thx

File: src/workspace/src/utils.ts (hunk 10 above)
Comment (algreasley) [reply src/workspace/src/utils.ts:13]: sorry, should have been more explicit about this .. I meant the "fx-" bit

Comment (algreasley) [reply src/workspace/src/utils.ts:13]: as opposed to the RT_URL/credit-* stuff

**Review by algreasley — APPROVED** [review]

**Pull request merged**